src/runtime/classbase.cs

```diff
@@ -72,6 +82,42 @@ public static IntPtr tp_richcompare(IntPtr ob, IntPtr other, int op)
         {
             CLRObject co1;
             CLRObject co2;
+            IntPtr tp = Runtime.PyObject_TYPE(ob);
+            var cls = (ClassBase)GetManagedObject(tp);
+            // C# operator methods take precedence over IComparable.
+            // We first check if there's a comparison operator by looking up the richcompare table,
+            // otherwise fallback to checking if an IComparable interface is handled.
+            if (PyToCilOpMap.ContainsKey(op)) {
+                string CilOp = PyToCilOpMap[op];
+                if (cls.richcompare.Contains(CilOp)) {
+                    var methodObject = (MethodObject)cls.richcompare[CilOp];
+                    IntPtr args = other;
+                    var free = false;
+                    if (!Runtime.PyTuple_Check(other))
```

Comment:
Indeed if `b` were a tuple this `if` would be skipped. Once removing the `if`, the operator method does receive  `b`  as `PyObject`,  but I get an `AccessViolationException : Attempted to read or write protected memory.` so I think `b` is wrongly parsed?

![image](https://user-images.githubusercontent.com/2038751/104018999-aa697a80-51b2-11eb-9534-9f2ec184dd67.png)

Reply:
I think that error message might have been a flaky bug, because now the variable states are the same but the error is:
```
Message: 
    Python.Runtime.PythonException : TypeError : '>=' not supported between instances of 'int' and 'tuple'
```

Still it means the `PyObject` was not recognized as a tuple

Reply:
This is because to work with `PyObject` instances in C# callbacks you must hold GIL. E.g. the method body must be inside `using (Py.GIL()) { ...` block.

Reply:
Thanks, I didn't know about that. https://github.com/pythonnet/pythonnet/blob/d6c0081f87b67707b59598efe877378b0c36d822/src/runtime/methodbinder.cs#L847 

![image](https://user-images.githubusercontent.com/2038751/104162736-1505fa80-53ed-11eb-8dc9-1a308938458e.png)


I also put the `operator >(Obj a, PyObject b)` body in a `Py.GIL` block but that doesn't do anything.

Reply:
I don't think you need to add a test with a tuple for each comparison operator. One is enough.

Reply:
The failure on the screenshot is only happening because you have debugger attached. It tries to call `ToString` on some `PyObject` to show in watches, but it does not work without `Py.GIL`